.github/workflows/create-tag-with-release-pr.yml

```
@@ -0,0 +1,64 @@
+name: Create tag and release on release PR merge
+on:
+  pull_request:
+    types:
+     - closed
+
+jobs:
+  run-release-trigger:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true && contains(github.event.pull_request.labels.*.name, 'release')
+    env:
+      GH_REF: ${{ github.event.pull_request.head.ref }}
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Prepare release tag version
+        id: prepare-tag
+        run: echo "PR_TAG_REF=$GH_REF" | sed "s/release-//" >> $GITHUB_OUTPUT
+      - name: Create tag
+        uses: actions/github-script@v5
+        with:
+          script: |
+            github.rest.git.createRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: "refs/tags/${{ steps.prepare-tag.outputs.PR_TAG_REF }}",
+              sha: context.sha
+            })
```

`event.pull_request.head.ref` is considered untrusted and dangerous as it carries user inputs[^1], use environment variables instead:

```suggestion
      - name: Create tag
        uses: actions/github-script@v5
        env:
          PR_REF: ${{ steps.prepare-tag.outputs.PR_TAG_REF }}
        with:
          script: |
            const { env } = require('node:process');

            github.rest.git.createRef({
              owner: context.repo.owner,
              repo: context.repo.repo,
              ref: "refs/tags/${env.PR_REF}",
              sha: context.sha
            })
```

[^1]: https://securitylab.github.com/resources/github-actions-untrusted-input/

---

@NyanKiyoshi the only place where we use untrusted `github.event.pull_request.head.ref` input is here:
https://github.com/saleor/saleor/blob/experimental-release-flow/.github/workflows/create-tag-with-release-pr.yml#L12-L17

It's already passed as env varable as described in https://securitylab.github.com/resources/github-actions-untrusted-input/ so do we need to repeat this action for all `PR_TAG_REF` usages?

---

Yes, the `PR_TAG_REF` needs to be used as an environment variable as it's a tainted user-input